lib/wasix/src/runtime/package_loader/load_package_tree.rs

```diff
@@ -180,7 +180,54 @@ fn load_binary_command(
         )
     })?;
 
-    let cmd = BinaryPackageCommand::new(command_name.to_string(), cmd.clone(), atom, hash);
+    // Get WebAssembly features from manifest atom annotations
+    let features = if let Some(atom_metadata) = webc.manifest().atoms.get(&atom_name) {
+        if let Ok(Some(wasm_annotation)) = atom_metadata
+            .annotation::<webc::metadata::annotations::Wasm>(webc::metadata::annotations::Wasm::KEY)
+        {
+            tracing::debug!(
+                "Found wasm features in atom annotation: {:?}",
+                wasm_annotation
+            );
+
+            // Create features object from the annotation strings
+            let mut features = wasmer_types::Features::default();
+
+            // Set features based on the string values in the annotation
+            features.simd(wasm_annotation.features.contains(&"simd".to_string()));
+            features.bulk_memory(
+                wasm_annotation
+                    .features
+                    .contains(&"bulk-memory".to_string()),
+            );
+            features.reference_types(
+                wasm_annotation
+                    .features
+                    .contains(&"reference-types".to_string()),
```

Comment:
This should be abstracted out, on the same layer that we are doing write/read.

Basically we should have two functions:

* `Features` -> Wasm annotations
And the reverse
*  Wasm annotations -> `Features`

Both living on the same space. One should be used when building a webc binary. Other when reading from a webc binary (aka here)

Reply:
IMO only Wasm annotations -> Features is needed. The otherway around is only used once.

Reply:
Even if is used once, having logic in two different places can become confusing and is good to tie them both together in one file

Reply:
Should be OK now.